python/activator/middleware_interface.py

```diff
@@ -165,9 +168,6 @@ def _init_local_butler(self, butler: Butler):
                        ]
         butler.registry.setCollectionChain(self.output_collection, collections)
 
-        # Refresh butler after configuring it, to ensure all required
-        # dimensions and collections are available.
-        butler.registry.refresh()
         self.butler = Butler(butler=butler,
```

Comment:
I know it's no longer a refresh, but it might be worth a comment her to say why we make a new butler instance for `self`, instead of using the one that was passed in.